ObjectOrientedProgramming/Memory Ownership/unique_ptr/task.md

```diff
@@ -0,0 +1,50 @@
+C++ contains a tool that provides exclusive ownership semantics. [std::unique_ptr](https://en.cppreference.com/w/cpp/memory/unique_ptr) is a "smart" pointer that, like all other smart pointers, is located in the `<memory>` library (so don't forget to connect it), and is designed to manage the lifetime of dynamically allocated objects. The unique ownership model ensures that at any given time, only one `std::unique_ptr` instance owns a particular dynamically allocated object. When the owning `std::unique_ptr` is destroyed or explicitly reset, it ensures the deallocation of the associated memory.
+
+#### Main features about std::unique_ptr ownership model
+- Exclusive ownership. When a dynamically allocated object comes into the possession of `std::unique_ptr`, no `std::unique_ptr` or other smart pointer (which we will cover further) can share ownership of this object.
+- The transfer of ownership takes place through move semantics (they will be discussed in the next step). When a `std::unique_ptr` is moved, the ownership is transferred, and the source `std::unique_ptr` becomes empty (`nullptr`).
+- Automatic Deallocation. When a `std::unique_ptr` goes out of scope or is explicitly reset, it automatically releases the memory it owns.
+- Custom Deleter Support. `std::unique_ptr` allows users to specify custom deleters. A deleter is a function or callable object responsible for releasing the memory when the `std::unique_ptr` is destroyed. This feature is useful for managing resources other than raw memory, such as file handles or database connections.
+- Support the correct state. `std::unique_ptr` is movable but not copyable. This aligns well with the ownership model, as copying would violate the exclusive ownership semantics.
+- Nullability. `std::unique_ptr` can be in a null state (pointing to no object) by default or after a move operation. This feature allows for representing the absence of an object in a safe and clear way.
+
+#### Benefits of using std::unique_ptr over pointers
```

Comment:
Please, do not overuse the bullets and lists. 
Make the text more interconnected, with a sequential flow of information. 
And interleave it with the examples (instead of just giving one single example at the end). 

You can have a look at the theory tasks in the module 2 for reference.